packages/trpc-playground/src/get-default-input.ts

```diff
@@ -0,0 +1,224 @@
+import { ZodAny } from 'zod'
+import { printNode, zodToTs } from 'zod-to-ts'
+
+type GenerateInput = {
+  operationType: string
+  operationName: string
+}
+
+// Generate code snippet for operation
+export const generateSnippet = (inputParser: ZodAny, { operationType, operationName }: GenerateInput) => {
+  const hasInput = !!inputParser?._def
+
+  let output: string = `await ${operationType}`
+  const input = hasInput ? getDefaultInput(inputParser) : ''
+  const type = hasInput ? getOperationType(inputParser) : ''
+
+  output += `('${operationName}'${input})`
+
+  return {
+    type,
+    default: { value: output, inputLength: input.length },
+  }
+}
+
+const getOperationType = (inputParser: ZodAny) => {
+  const { node } = zodToTs(inputParser)
+  return printNode(node)
+}
+
+// Get default input string for query/mutation defined in the router
+const getDefaultInput = (inputParser: ZodAny): string => {
+  if (typeof inputParser === 'function') return ''
+
+  const defaultInput = getDefaultForDef(inputParser._def)
+  return defaultInput ? `, ${defaultInput}` : ``
+}
+
+import {
+  ZodArrayDef,
+  ZodEnumDef,
+  ZodFirstPartyTypeKind,
+  ZodIntersectionDef,
+  ZodLiteralDef,
+  ZodMapDef,
+  ZodNativeEnumDef,
+  ZodNullableDef,
+  ZodNumberDef,
+  ZodObjectDef,
+  ZodOptionalDef,
+  ZodPromiseDef,
+  ZodRecordDef,
+  ZodSetDef,
+  ZodStringDef,
+  ZodTupleDef,
+  ZodTypeDef,
+  ZodUnionDef,
+} from 'zod'
+
+const getDefaultForDef = (def: any): string | undefined => {
+  if (!def) return
+
+  switch ((def as any).typeName) {
+    case ZodFirstPartyTypeKind.ZodString:
+      return defaultString(def)
```

Comment:
what if the `typeName` is none of these cases?

Reply:
Maybe we could just return an empty string?

Reply:
sounds good to me, I think this is what was already happening actually